### DISCUSSION TITLE
PFC security built into utility

### DISCUSSION THREAD

**Discussion by jojo5791** [OpenSourcePFCLibraries/2022#5] [Q&A] [answered]
Hi Guys,
  WOuld need your help to learn how to build the pfc user security into utility. I setup as per picture attached. But scanner always give me errors when full build it. After compile and build into exe, it didnt run after i choose 'Scan Application' i mean, the _targetselection window didn show. I got the second picture when i try to  compile it. Please help me if have miss anything? thanks in advance.

![image](https://github.com/OpenSourcePFCLibraries/2022/assets/70992666/ab959d0e-ade9-4ef9-87a9-71c41479fc62)
![image](https://github.com/OpenSourcePFCLibraries/2022/assets/70992666/427740fb-70e1-466d-a333-3711d40cc336)

**Comment by bruce-armstrong** [comment]
We’ve got your messages.  This is an open source project, which means it’s supported by developers like yourself that have their own full time jobs.  I’ll try to look into this, but it likely won’t be until sometime this weekend.

 

From: JaneD ***@***.***> 
Sent: Friday, September 15, 2023 8:05 AM
To: OpenSourcePFCLibraries/2022 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [OpenSourcePFCLibraries] PFC security built into utility (Discussion #5)

 

Hi Guys,
WOuld need your help to learn how to build the pfc user security into utility. I setup as per picture attached. But scanner always give me errors when full build it. After compile and build into exe, it didnt run after i choose 'Scan Application' i mean, the _targetselection window didn show. I got the second picture when i try to compile it. Please help me if have miss anything? thanks in advance.

 <https://user-images.githubusercontent.com/70992666/268319492-ab959d0e-ade9-4ef9-87a9-71c41479fc62.png> 
 <https://user-images.githubusercontent.com/70992666/268320066-427740fb-70e1-466d-a333-3711d40cc336.png>

**Comment by bruce-armstrong** [comment, marked as answer]
Well, due to getting ready for Elevate and following up on items coming out of it that took a lot longer than I expected.  Apologize for that.

I think the primary issue is that you've added the PFC libraries to the library search path.  The security scanner is *not* a PFC based application.  In a PFC based application there would be a global variable of type n_cst_appmanager or a descendant of that.  Also, the SQLCA, error and message variable types for the application object would likely be based off of n_trans, n_err and n_msg

The errors you're seeing is from the PFC layer not finding that global variable app manager, which isn't needed for the utility.

Try removing the PFC libraries from that target and recompiling it.  It works fine for me.

![image](https://github.com/OpenSourcePFCLibraries/2022/assets/6356830/3b89c0ac-403e-47df-be37-6872923aad58)
